Examples/SlackClone/SlackComposerVC.swift

```diff
@@ -94,10 +94,12 @@ open class ComposerEmojiSuggestionsDataSource: NSObject, UICollectionViewDataSou
     }
 
     public func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCell(
+        guard let cell = collectionView.dequeueReusableCell(
             withReuseIdentifier: ComposerEmojiSuggestionsCollectionViewCell.reuseId,
             for: indexPath
-        ) as! ComposerEmojiSuggestionsCollectionViewCell
+        ) as? ComposerEmojiSuggestionsCollectionViewCell else {
+            return UICollectionViewCell()
+        }
```

Comment:
How do you feel about introducing `Reusable` type for both table & collection views? 🙂 The iOS community is familiar with this approach and for us, it would allow to remove the boilerplate and make code easy to read.

```swift
protocol Reusable {
    static var reuseIdentifier: String { get }
}

extension Reusable {
    static var reuseIdentifier: String {
        String(describing: self)
    }
}

extension UITableView {
    func register<Cell: UITableViewCell>(_ type: Cell.Type) where Cell: Reusable {
        register(type, forCellReuseIdentifier: type.reuseIdentifier)
    }
    
    func dequeueReusableCell<Cell: UITableViewCell>(_ type: Cell.Type, for indexPath: IndexPath) -> Cell where Cell: Reusable {
        // here it's acceptable to disable the force-case rule, i think
        return dequeueReusableCell(withIdentifier: type.reuseIdentifier, for: indexPath) as! Cell
    }
}
```

Reply:
Yeah, it's a good question - I personally would be happy with this... my biggest question really was what is the ideal approach here should we force unwrap in general on cells and if the answer is no then do we just return empty cell or something else (log, assert, etc...)

Reply:
Yeah, that's not a conditional failure, that's a dev mistake if the force case of the cell fails so someone checking the DemoApp should never experience these error (unless they don't change sources). I honestly think `as!` was fine but if we encapsulate it in the nice new dequeue API we apply the logging for all dequeues as once.

func dequeueReusableCell<Cell: UITableViewCell>(_ type: Cell.Type, for indexPath: IndexPath) -> Cell where Cell: Reusable {
        // here it's acceptable to disable the force-case rule, i think
        guard let cell = dequeueReusableCell(withIdentifier: type.reuseIdentifier, for: indexPath) as? Cell else {
         log.error()
         return Cell.init()
        }
        return cell
    }

Reply:
I like the above suggestion from @evsaev